src/main/java/edu/ie3/util/io/FileIOUtils.java

```diff
@@ -141,6 +157,495 @@ public static BufferedWriter getBufferedWriter(
+
+        /* Control the size of extracted archive files */
+        long uncompressedSize = archiveEntry.getSize();
+        if (uncompressedSize == ArchiveEntry.SIZE_UNKNOWN)
+          throw new IOException(
+              "Unknown uncompressed file size of '" + archiveEntry.getName() + "'");
+        size += uncompressedSize;
+        if (size > MAX_SIZE_UNCOMPRESSED)
+          throw new IOException(
+              "Uncompressed size of archive exceeds permissible "
+                  + (MAX_SIZE_UNCOMPRESSED / 1024 / 1024)
+                  + " MB. Possibly malicious archive");
+
+        /* Control the compression ratio */
+        if (1 - (double) archiveSize / size > MAX_COMPRESSION_RATIO)
+          throw new IOException(
+              "Compression ratio exceeds its maximum permissible value "
+                  + (MAX_COMPRESSION_RATIO * 100)
+                  + " %. Possibly malicious archive");
+
+        handleZipEntrySafely(archiveEntry, targetDirectory, tarInputStream);
+      }
+    } catch (IOException ex) {
+      throw new FileException("Unable to extract from '" + archive + "'.", ex);
+    }
+
+    return targetDirectory;
+  }
+
+  /**
+   * Extracts the given zipped file to a file with the same name that the zipped file has beneath
+   * the target directory.
+   *
+   * @param zippedFile Compressed gzip file to extract
+   * @param target Path to the target folder
+   * @return Path of the file, where the content is extracted to
+   * @throws FileException If the zipped file is not in a well shape or the target path doesn't meet
+   *     the requirements
+   * @see <a href="https://mkyong.com/java/how-to-decompress-file-from-gzip-file/">MyKong
+   *     Tutorial</a>
+   */
+  public static Path extractFile(Path zippedFile, Path target) throws FileException {
+    /* Pre-flight checks and assembly of the target path */
+    Path targetPath = validateTargetAndZippedFile(zippedFile, target);
+
+    /* Get the zipped file size */
+    long zippedSize = zippedFile.toFile().length();
+
+    /* Create the destination folder for unzipping the zip file */
+    try {
+      Path parentDirectoryPath = targetPath.getParent();
+      if (parentDirectoryPath != null) {
+        if (Files.notExists(parentDirectoryPath)) {
+          Files.createDirectories(parentDirectoryPath);
+        }
+      } else {
+        throw new FileException("Parent directory path is null for the file '" + targetPath + "'.");
+      }
+    } catch (IOException e) {
+      throw new FileException("Cannot create target folder for the file '" + targetPath + "'.", e);
+    }
+
+    try (InputStream fileInputStream = new FileInputStream(zippedFile.toFile());
+        GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream); ) {
+
+      long uncompressedSize = 0;
+
+      /*Create a temporary file for extracting contents of the zipped file*/
+      File tempFile = File.createTempFile("temp", null);
```

Comment:
I'm not sure, if it is necessary to create a temporary file, as this imposes the need to copy and delete it. Would it be an option to directly write to the target file and delete it, if an exception is thrown?

Reply:
Yes, we can do it. I will make the changes.